src/form/remaining-chars-count/remaining-chars-count.directive.js

```diff
@@ -0,0 +1,177 @@
+                         count-fld="charRemainingCntFld_1" name="text" placeholder="Type in your message" rows="5"></textarea>
+             </div>
+             <span class="pull-right chars-remaining-pf">
+               <span id="charRemainingCntFld_1"></span>
+               <button id="postBtn_1" ng-disabled="charsMaxLimitExceeded" type="submit" class="btn btn-default">Post New Message</button>
+             </span>
+           </form>
+
+         </div>
+       </div>
+       <br>
+       <strong>Max Limit: 10, Warn When 2 Or less Remaining, Block Input After Max Limit</strong>
+       <div class="row">
+         <div class="col-md-4">
+          <form>
+             <div class="form-group">
+               <label for="messageArea"></label>
+               <textarea class="form-control" pf-remaining-chars-count id="messageArea_2" ng-model="messageArea2text" chars-max-limit="10" chars-warn-remaining="2"
+                         block-input-at-max-limit="true" count-fld="charRemainingCntFld_2" name="text"
+                         placeholder="Type in your message" rows="5"></textarea>
+             </div>
+             <span class="pull-left">
+               <button id="postBtn_2" type="submit" class="btn btn-default">Submit</button>
+             </span>
+             <span class="pull-right chars-remaining-pf">
+               <span id="charRemainingCntFld_2"></span>
+             </span>
+           </form>
+         </div>
+       </div>
+       <br>
+       <strong>Max Limit: 10, Warn When 5 Or less Remaining, Block Input After Max Limit</strong>
+       <div class="row">
+         <div class="col-md-4">
+           <input id="input_3" pf-remaining-chars-count chars-max-limit="10" ng-model="messageInput3text" chars-warn-remaining="5" count-fld="charRemainingCntFld_3"
+             block-input-at-max-limit="true"/>
+             <span class="chars-remaining-pf"><span id="charRemainingCntFld_3" style="padding-left: 5px"></span>Remaining</span>
+         </div>
+       </div>
+     </div>
+   </file>
+
+   <file name="script.js">
+   angular.module( 'patternfly.example', ['patternfly.form']);
+
+   angular.module( 'patternfly.example' ).controller( 'DemoCtrl', function( $scope ) {
+     $scope.messageArea1text = "Initial Text";
+     $scope.messageArea2text = "";
+     $scope.messageInput3text = "";
+
+     $scope.charsMaxLimitExceeded = false;
+
+     // 'tfId' will equal the id of the text area/input field which
+     // triggered the event
+     $scope.$on('overCharsMaxLimit', function (event, tfId) {
+         if(!$scope.charsMaxLimitExceeded){
+           $scope.charsMaxLimitExceeded = true;
+           $scope.$digest();
+         }
+     });
+
+     // 'tfId' will equal the id of the text area/input field which
+     // triggered the event
+     $scope.$on('underCharsMaxLimit', function (event, tfId) {
+         if($scope.charsMaxLimitExceeded){
+           $scope.charsMaxLimitExceeded = false;
+           $scope.$digest();
```

Comment:
In those situations where there might be multiple textareas, it might be good to know which one dispatched the event.  In case they are listening on the body.